lib/registry.rb

```diff
@@ -0,0 +1,119 @@
+require "timed_cache"
+
+module Registry
+  class BaseRegistry
+    CACHE_LIFETIME = 12 * 3600  # 12 hours
+
+    def initialize(index, format, fields = %w{slug link title}, clock = Time)
+      @cache = TimedCache.new(CACHE_LIFETIME, clock) { fetch }
+      @fields = fields
+      @format = format
+      @index = index
+    end
+
+    def all
+      @cache.get
+    end
+
+    def [](slug)
+      all.find { |o| o.slug == slug }
+    end
+
+  private
+    def fetch
+      find_documents_by_format.to_a
+    end
+
+    def find_documents_by_format
+      @index.documents_by_format(@format, fields: @fields)
+    end
+  end
+
+  class DocumentCollection < BaseRegistry
+    def initialize(index)
+      super(index, "document_collection")
+    end
+  end
+
+  class DocumentSeries < BaseRegistry
+    def initialize(index)
+      super(index, "document_series")
+    end
+  end
+
+  class Organisation < BaseRegistry
+    def self.load_ministerial_departments
+      file_path = File.join(File.dirname(__FILE__), "..", "config", "ministerial_departments.txt")
+      File.readlines(file_path).map(&:chomp).reject(&:empty?).map do |slug|
+        "/government/organisations/#{slug}"
+      end
+    end
+
+    MINISTERIAL_DEPARTMENT_LINKS = load_ministerial_departments
+    MINISTERIAL_DEPARTMENT_TYPE = "Ministerial department"
+
+    def initialize(index)
+      super(index, "organisation", %w{slug link title acronym organisation_type organisation_state})
+    end
+
+  private
+    def fetch
+      find_documents_by_format.map { |o| fill_organisation_type(o) }
+    end
+
+    def fill_organisation_type(organisation)
+      # If the organisation type isn't indexed, populate it if we know the
+      # organisation is a ministerial department.
+      if organisation.has_field?(:organisation_type) && organisation.organisation_type.nil?
+        if MINISTERIAL_DEPARTMENT_LINKS.include?(organisation.link)
+          organisation.organisation_type = MINISTERIAL_DEPARTMENT_TYPE
+        end
+      end
+
+      organisation
+    end
+  end
+
+  class SpecialistSector < BaseRegistry
+    CACHE_LIFETIME_SECONDS = 300
```

Comment:
This constant has a different name to the one on the base class, meaning it won;t have any effect.  Additionally, I don't think it would apply even if the name did match due to scoping.  I'll have a look at fixing this.